methods that implement objc protocol don't seem to be called

While trying to port https://github.com/TextureGroup/Texture/blob/master/examples/ASViewController/Sample/ViewController.m I ended up with the following code:
```
class MyViewController protected constructor(
    val tableNode: ASTableNode
) : ASViewController(tableNode), ASTableDataSourceProtocol {

    val imageCategories: Array<String> = arrayOf("abstract", "animals", "business")

    init {
        NSLog("my init")
    }

    @OverrideInit
    constructor(): this(ASTableNode(UITableViewStyle.UITableViewStylePlain)) {
        NSLog("my constructor")
    }

    override fun viewDidLoad() {
        NSLog("my viewDidLoad")

        super.viewDidLoad()

        tableNode.dataSource = this
        tableNode.layoutIfNeeded()
    }

    override fun numberOfSectionsInTableNode(tableNode: ASTableNode): Long {
        NSLog("my numberOfRowsInSection: 1")
        return 1
    }

    override fun tableNode(tableNode: ASTableNode, numberOfRowsInSection: Long): Long {
        val rows = imageCategories.count().toLong()

        NSLog("my numberOfRowsInSection: $rows")
        return rows
    }

    override fun tableView(tableView: ASTableView, nodeBlockForRowAtIndexPath: NSIndexPath): ASCellNodeBlock? {
        val category = imageCategories[nodeBlockForRowAtIndexPath.row.toInt()]

        NSLog("my nodeBlockForRowAtIndexPath: $category")
        return {
            val cell = ASTextCellNode()
            cell.text = category
            cell
        }
    }

    override fun debugDescription(): String? {
        return super<ASViewController>.debugDescription()
    }
}
```

When I navigate to my view controller, here is what gets logged:
```
2018-09-04 15:50:57.320747+0400 konan[62352:14952406] my constructor
2018-09-04 15:50:57.321814+0400 konan[62352:14952406] my viewDidLoad
```

A few questions on that:
1. Is kotlin's `init {}` supposed to be called?
2. Non of the methods of `ASTableDataSourceProtocol` are called. Am I supposed to mark them with an annotation?

I added more logs and found out some weirdness:

```
class MyViewController protected constructor(
    val tableNode: ASTableNode
) : ASViewController(tableNode), ASTableDataSourceProtocol {

    val imageCategories: Array<String> = arrayOf("abstract", "animals", "business")

    init {
        NSLog("my init")
    }

    @OverrideInit
    constructor(): this(ASTableNode(UITableViewStyle.UITableViewStylePlain)) {
        NSLog("my constructor")
    }

    override fun viewDidLoad() {
        NSLog("my viewDidLoad")

        super.viewDidLoad()

        if (node == null) {
            NSLog("node is null!")
        }

        if (tableNode == null) {
            NSLog("tableNode is null!")
        }

        tableNode.dataSource = this
        tableNode.layoutIfNeeded()
    }

    override fun numberOfSectionsInTableNode(tableNode: ASTableNode): Long {
        NSLog("my numberOfRowsInSection: 1")
        return 1
    }

    override fun tableNode(tableNode: ASTableNode, numberOfRowsInSection: Long): Long {
        val rows = imageCategories.count().toLong()

        NSLog("my numberOfRowsInSection: $rows")
        return rows
    }

    override fun tableView(tableView: ASTableView, nodeBlockForRowAtIndexPath: NSIndexPath): ASCellNodeBlock? {
        val category = imageCategories[nodeBlockForRowAtIndexPath.row.toInt()]

        NSLog("my nodeBlockForRowAtIndexPath: $category")
        return {
            val cell = ASTextCellNode()
            cell.text = category
            cell
        }
    }

    override fun debugDescription(): String? {
        return super<ASViewController>.debugDescription()
    }
}
```

The log:
```
2018-09-04 17:36:21.277397+0400 konan[66571:15140142] my constructor
2018-09-04 17:36:21.278526+0400 konan[66571:15140142] my viewDidLoad
2018-09-04 17:36:21.278736+0400 konan[66571:15140142] node is null!
2018-09-04 17:36:21.278916+0400 konan[66571:15140142] tableNode is null!
```

Given that `tableNode` is not optional, how come it's null?

I even get compiler warning:
```
src/main/kotlin/main.kt:78:13: warning: condition 'tableNode == null' is always 'false'
        if (tableNode == null) {
```

Guys, do you have any ideas what goes wrong?

--- a/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt
+++ b/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt
@@ -2583,7 +2583,8 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
         val constructedClass = functionGenerationContext.constructedClass!!
         val thisPtr = currentCodeContext.genGetValue(constructedClass.thisReceiver!!)
 
-        if (constructedClass.isObjCClass()) {
+        if (descriptor.constructedClass.isExternalObjCClass()) {
+            assert(args.isEmpty())
             return codegen.theUnitInstanceRef.llvm
         }
 

--- a/backend.native/tests/interop/objc/smoke.kt
+++ b/backend.native/tests/interop/objc/smoke.kt
@@ -45,6 +45,10 @@ fun run() {
     pair.swap()
     println("${pair.first}, ${pair.second}")
 
+    val defaultPair = MutablePairImpl()
+    assertEquals(defaultPair.first(), 123)
+    assertEquals(defaultPair.second(), 321)
+
     // equals and hashCode (virtually):
     val map = mapOf(foo to pair, pair to foo)
 
@@ -105,6 +109,8 @@ class MutablePairImpl(first: Int, second: Int) : NSObject(), MutablePairProtocol
     override fun update(index: Int, sub: Int) {
         elements[index] -= sub
     }
+
+    constructor() : this(123, 321)
 }
 
 fun testTypeOps() {

